test/e2e/servicebindingrequest_test.go

```diff
@@ -285,3 +300,108 @@ func serviceBindingRequestTest(t *testing.T, ctx *framework.TestCtx, f *framewor
 	_ = f.Client.Delete(todoCtx, sbrSecret)
 	_ = f.Client.Delete(todoCtx, &d)
 }
+
+// This test runs with following steps (scenario-2)
+// 1. Create SBR.
+// 2. Create operator-backed database CR.
+// 3. Create deployment where SBR would inject environment variables.
+// 4. Run assertion
+
+func preCreateServiceBindingRequestTest(t *testing.T, ctx *framework.TestCtx, f *framework.Framework, ns string) {
```

Comment:
I like the idea of adding one more scenario to the end-to-end testing, that's good move! However, I see that we are mostly copying and pasting the code, and I think we should try to extract as much as possible to functions, so we can re-use those to create more test case scenarios in the near future.